src/encoded/audit/biosample.py

```diff
@@ -61,6 +61,31 @@ def audit_biosample_modifications(value, system):
 # https://encodedcc.atlassian.net/browse/ENCD-3538
 
 
+def audit_biosample_CRISPR_modifications(value, system):
+# flag biosamples with multiple CRISPR/characterization GM; may be legitimate but we want be notified
+# https://encodedcc.atlassian.net/browse/ENCD-5203
+    if 'applied_modifications' in value:
+        if len(value['applied_modifications']) > 1:
+            method = 0
+            purpose = 0
+            GM_ids = set()
+            for GM in value['applied_modifications']:
+                if GM['method'] == 'CRISPR':
+                    method += 1
+                    if GM['purpose'] == 'characterization':
+                        purpose += 1
+                        GM_ids.add(GM['@id'])
+            GM_ids_links = [audit_link(path_to_text(m), m) for m in GM_ids]
+            if purpose == method & purpose >1:
```

Comment:
I think this is not going to work in cases where you have (CRISPR-characterization, CRISPR-characterization, CRISPR-not_characterization) method will be 3 and purpose will be 2 so despite 2 CRISPR characterization the audit will not trigger?

Reply:
Somehow it does seem to work correctly (I checked on the demo), but rereading the code I'm not sure how that is possible. I think I see now how to write this in a way that's more clean and robust, thanks!